src/main/java/uk/gov/di/ipv/cri/common/library/service/SessionService.java

```diff
@@ -175,9 +191,25 @@ public SessionItem getSessionByAuthorisationCode(String authCode)
         sessionItem = validateSessionId(String.valueOf(sessionItem.getSessionId()));
 
         if (sessionItem.getAuthorizationCodeExpiryDate() < clock.instant().getEpochSecond()) {
+            LOGGER.error(
+                    "Access Token could not be issued. The supplied authorization code has expired. Expired at: {}",
+                    sessionItem.getAuthorizationCodeExpiryDate());
             throw new AuthorizationCodeExpiredException("authorization code expired");
         }
 
         return sessionItem;
     }
+
+    public void revokeAccessToken(SessionItem sessionItem) throws IllegalArgumentException {
```

Comment:
there is also a revokeAccesstoken method in AccessTokenService.
Therefore, it would probably make sense to remove this method and use the AccessTokenService method followed by a call to SessionService.updateSession